### PR TITLE
chore(deps): Upgrade to eslint 6

### PR DESCRIPTION
**Pull request by blowery**
#### Changes proposed in this Pull Request

* Upgrade to eslint 6, including all plugins and configs

#### Testing instructions

* `npm run lint:js` should still work

### CODE REVIEW

File: apps/notifications/src/panel/templates/actions.jsx [hunk 1]
```diff
@@ -31,14 +31,14 @@ const getInitialReplyValue = ( note, translate ) => {
 	}
 
 	if ( username ) {
-		return translate( 'Reply to %(username)s...', {
+		return translate( 'Reply to %(username)s…', {
```
Comment (a8ci18n) [inline apps/notifications/src/panel/templates/actions.jsx]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

File: apps/notifications/src/panel/templates/actions.jsx [hunk 2]
```diff
@@ -31,14 +31,14 @@ const getInitialReplyValue = ( note, translate ) => {
...
 			args: { username },
 		} );
 	}
 
 	return getType( note ) === 'post'
-		? translate( 'Reply to post...' )
-		: translate( 'Reply to comment...' );
+		? translate( 'Reply to post…' )
```
Comment (a8ci18n) [inline apps/notifications/src/panel/templates/actions.jsx]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

File: client/lib/track-element-size/index.ts [hunk 3]
```diff
@@ -76,7 +76,7 @@ export function useWindowResizeCallback(
 			window.removeEventListener( 'resize', throttledMeasureElement );
 			throttledMeasureElement.cancel();
 		};
-	}, [ elementRef.current, callback ] );
+	}, [ callback ] );
```
Comment (blowery) [inline client/lib/track-element-size/index.ts]: eslint@6 "fixed" this, but it seems to be wrong? at least it broke the test...

File: client/lib/user-settings/test/index.js [hunk 4]
```diff
@@ -18,19 +18,21 @@ describe( 'User Settings', () => {
 		userSettings.fetchSettings();
 	} );
 
-	test( 'should consider overridden settings as saved', done => {
-		expect( userSettings.updateSetting( 'test', true ) ).toBe( true );
-		expect( userSettings.updateSetting( 'lang_id', true ) ).toBe( true );
-		expect( userSettings.unsavedSettings.test ).toBe( true );
-		expect( userSettings.unsavedSettings.lang_id ).toBe( true );
+	test( 'should consider overridden settings as saved', () => {
```
Comment (blowery) [inline client/lib/user-settings/test/index.js]: some of these new autofixes are _wild_

File: packages/eslint-plugin-wpcalypso/lib/rules/i18n-no-collapsible-whitespace.js [hunk 5]
```diff
@@ -32,7 +32,7 @@ const rule = ( module.exports = function( context ) {
 					return;
 				}
 
-				const collapsibleWhitespace = string.match( /(\n|\t|\r|(?:  ))/ );
+				const collapsibleWhitespace = string.match( /(\n|\t|\r|(?: {2}))/ );
```
Comment (blowery) [inline packages/eslint-plugin-wpcalypso/lib/rules/i18n-no-collapsible-whitespace.js]: whoa.

File: client/lib/track-element-size/index.ts (hunk 3 above)
Comment (sgomes) [reply client/lib/track-element-size/index.ts]: Right, I don't think that's a valid fix. It seems you dropped it for now, though?

Comment (blowery) [reply client/lib/track-element-size/index.ts]: yup, dropped for now. curious what it'll do with the v2 of the hooks plugin.

**Comment by sirreal** [comment]
I just pushed a commit to use [`parser-options`](https://eslint.org/docs/user-guide/command-line-interface#--parser-options) for the es5 validator which seems to be the way to do it for eslint@6.

**Comment by blowery** [comment]
@sirreal fancy, thanks!

File: client/lib/track-element-size/index.ts (hunk 3 above)
Comment (blowery) [reply client/lib/track-element-size/index.ts]: @sgomes would you mind trying https://github.com/Automattic/wp-calypso/pull/37404 and see if that changes how it "fixes" things?

**Review by sirreal — APPROVED** [review]
👍 I don't see any reason not to go ahead with this.

File: client/lib/track-element-size/index.ts (hunk 3 above)
Comment (sgomes) [reply client/lib/track-element-size/index.ts]: As far as I can tell, there's no fix anymore (assuming my IDE is picking it up correctly). The code I wrote is probably incorrect, though; I'll need to take a closer look at this.

**Pull request merged**